ext/search_kit/Civi/Api4/Action/SearchDisplay/GetSearchTasks.php

```diff
@@ -138,15 +138,30 @@ public function _run(\Civi\Api4\Generic\Result $result) {
             $task['title'] = E::ts('Profile Update');
           }
           $key = \CRM_Core_Key::get(\CRM_Utils_Array::first((array) $task['class']), TRUE);
-          $tasks[$entity['name']]['contact.' . $id] = [
-            'title' => $task['title'],
-            'icon' => $task['icon'] ?? 'fa-gear',
-            'crmPopup' => [
-              'path' => "'{$task['url']}'",
-              'query' => "{reset: 1}",
-              'data' => "{cids: ids.join(','), qfKey: '$key'}",
-            ],
-          ];
+
+          //break out mailing labels for redirect instead of crmPopup
+          if ($task['title'] == 'Mailing labels - print') {
```

Comment:
I haven't run this but I think it would be better to check the loop index $id, to be more translation-friendly. `if $id == CRM_Core_Task::LABEL_CONTACTS`

Reply:
good suggestion just updated the code